yaml/com/ibm/VPD/FRUManager.interface.yaml

```diff
@@ -0,0 +1,121 @@
+          - name: data
+            type:
+                variant[struct[string,string,array[byte]],
+                struct[string,array[byte]]]
+            description: >
+                Information required to update the value of a keyword and the
+                data to write.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: ReadKeyword
+      description: >
+          Method to read VPD keyword value. Keyword can be read from hardware or
+          from Dbus.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Dbus/EEPROM Path of the FRU whose keyword needs to be read.
+          - name: data
+            type: variant[struct[string,string],struct[string]]
+            description: >
+                Information required to read value of keyword.
+          - name: target
+            type: boolean
+            description: >
+                Default will be false and will read from hardware in that case,
+                if true will read from cache.
+      returns:
+          - name: value
+            type: array[byte]
+            description: >
+                Value of the given keyword read from the given target.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: GetLocationCode
+      description: >
+          Method to get the expanded location code for the given inventory path.
+          If expanded location code is not available then unexpanded location
+          code will be returned.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU.
+      returns:
+          - name: locationCode
+            type: string
+            description: >
+                Returns the expanded location code if available, else returns
+                the unexpanded location code.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: deleteFRUVPD
+      description: >
+          Method to delete given FRU VPD and set its present property as false.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: CollectFRUVPD
+      description: >
+          Method to perform VPD collection for the given FRU. This method
+          requires to have configuration JSON. The FRU should have concurrently
+          Maintainable flag set to true in JSON to execute this action.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU whose VPD needs to be collected.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: PerformVPDRecollection
```

Comment:
Why do we need this API exposed? Is there a use case of it getting called from outside?
If no, I guess we don't need this?

Reply:
It is not getting called from outside
Removed the API